files/tasks.py

```
@@ -1246,48 +1273,119 @@ def beat_test(x, y):
 def kill_ffmpeg_process(filepath):
-    # this is not ideal, ffmpeg pid could be linked to the Encoding object
+    """Kill all ffmpeg processes matching the given filepath.
+
+    pgrep can return multiple PIDs (one per line), so we split
+    and kill each individually.
+    """
+    if not filepath:
+        return None
     try:
-        # Use pgrep to find ffmpeg processes with the filepath
         result = subprocess.run(
```

Ruff (0.15.2)

[error] `subprocess` call: check for execution of untrusted input

(S603)